src/workerd/api/rtti.c++

```diff
@@ -0,0 +1,196 @@
+#include <workerd/api/html-rewriter.h>
+#include <workerd/api/kv.h>
+#include <workerd/api/modules.h>
+#include <workerd/api/queue.h>
+#include <workerd/api/r2.h>
+#include <workerd/api/r2-admin.h>
+#include <workerd/api/sockets.h>
+#include <workerd/api/scheduled.h>
+#include <workerd/api/sql.h>
+#include <workerd/api/streams/standard.h>
+#include <workerd/api/trace.h>
+#include <workerd/api/urlpattern.h>
+#include <workerd/api/node/node.h>
+#include <workerd/jsg/modules.capnp.h>
+
+#include <cloudflare/cloudflare.capnp.h>
+
+#define EW_TYPE_GROUP_FOR_EACH(F)                                              \
+  F("dom-exception", jsg::DOMException)                                        \
+  F("global-scope", EW_GLOBAL_SCOPE_ISOLATE_TYPES)                             \
+  F("durable-objects", EW_ACTOR_ISOLATE_TYPES)                                 \
+  F("durable-objects-state", EW_ACTOR_STATE_ISOLATE_TYPES)                     \
+  F("analytics-engine", EW_ANALYTICS_ENGINE_ISOLATE_TYPES)                     \
+  F("basics", EW_BASICS_ISOLATE_TYPES)                                         \
+  F("blob", EW_BLOB_ISOLATE_TYPES)                                             \
+  F("cache", EW_CACHE_ISOLATE_TYPES)                                           \
+  F("crypto", EW_CRYPTO_ISOLATE_TYPES)                                         \
+  F("encoding", EW_ENCODING_ISOLATE_TYPES)                                     \
+  F("form-data", EW_FORMDATA_ISOLATE_TYPES)                                    \
+  F("html-rewriter", EW_HTML_REWRITER_ISOLATE_TYPES)                           \
+  F("http", EW_HTTP_ISOLATE_TYPES)                                             \
+  F("kv", EW_KV_ISOLATE_TYPES)                                                 \
+  F("queue", EW_QUEUE_ISOLATE_TYPES)                                           \
+  F("r2-admin", EW_R2_PUBLIC_BETA_ADMIN_ISOLATE_TYPES)                         \
+  F("r2", EW_R2_PUBLIC_BETA_ISOLATE_TYPES)                                     \
+  F("scheduled", EW_SCHEDULED_ISOLATE_TYPES)                                   \
+  F("streams", EW_STREAMS_ISOLATE_TYPES)                                       \
+  F("trace", EW_TRACE_ISOLATE_TYPES)                                           \
+  F("url", EW_URL_ISOLATE_TYPES)                                               \
+  F("url-standard", EW_URL_STANDARD_ISOLATE_TYPES)                             \
+  F("url-pattern", EW_URLPATTERN_ISOLATE_TYPES)                                \
+  F("websocket", EW_WEBSOCKET_ISOLATE_TYPES)                                   \
+  F("sql", EW_SQL_ISOLATE_TYPES)                                               \
+  F("sockets", EW_SOCKETS_ISOLATE_TYPES)                                       \
+  F("node", EW_NODE_ISOLATE_TYPES)                                             \
+  F("rtti", EW_RTTI_ISOLATE_TYPES)
+
+namespace workerd::api {
+
+namespace {
+  struct EncoderErrorReporterImpl : public Worker::ValidationErrorReporter {
```

Comment:
super minor non-blocking style nit: stuff inside the anon namespace shouldn't be indented.